ibis/backends/base/sql/alchemy/datatypes.py

```diff
@@ -61,34 +56,46 @@ def compiles_map(element, compiler, **kw):
     return f"MAP({key_type}, {value_type})"
 
 
-class UInt64(sa.types.Integer):
+class UInt64(sat.Integer):
     pass
 
 
-class UInt32(sa.types.Integer):
+class UInt32(sat.Integer):
     pass
 
 
-class UInt16(sa.types.Integer):
+class UInt16(sat.Integer):
     pass
 
 
-class UInt8(sa.types.Integer):
+class UInt8(sat.Integer):
     pass
 
 
 @compiles(UInt64, "postgresql")
 @compiles(UInt32, "postgresql")
 @compiles(UInt16, "postgresql")
 @compiles(UInt8, "postgresql")
+@compiles(UInt64, "mssql")
+@compiles(UInt32, "mssql")
+@compiles(UInt16, "mssql")
+@compiles(UInt8, "mssql")
 @compiles(UInt64, "mysql")
 @compiles(UInt32, "mysql")
 @compiles(UInt16, "mysql")
 @compiles(UInt8, "mysql")
+@compiles(UInt64, "snowflake")
+@compiles(UInt32, "snowflake")
+@compiles(UInt16, "snowflake")
+@compiles(UInt8, "snowflake")
 @compiles(UInt64, "sqlite")
 @compiles(UInt32, "sqlite")
 @compiles(UInt16, "sqlite")
 @compiles(UInt8, "sqlite")
+@compiles(UInt64, "trino")
+@compiles(UInt32, "trino")
+@compiles(UInt16, "trino")
+@compiles(UInt8, "trino")
```

Comment:
I kept these here to avoid repeating this code for each of the backends that doesn't support unsigned integer types.

There are no hidden dependencies here, the dialects are never imported if this function is never called.